### PR TITLE
add system v init support

### PR DESCRIPTION
**Pull request by kmohrf**
this adds support for system v init, by adding a traditional init file based on the lsb init standard and the debian specific init-functions.

while browsing through the code i noticed that the current state of service handling is a little bit different depending on the detected init system.
1. systemd registers and starts the service
2. upstart only starts the service? (maybe it registers too, the upstart.conf has runlevels defined)
3. if service is detected it’s only started
4. new addition: if update-rc.d is detected service will be enabled and then started via /etc/init.d

maybe it’s good as is. i’m just not sure :). but that isn’t necessarily part of this pr.

### CODE REVIEW

**Comment by kmohrf** [comment]
actually… i think something has to be done. `elif hash service 2> /dev/null; then` now matches for debian too without registering the service. if upstart auto-registers via the runlevel definition it propbably should be something like:

1. check systemd && register via systemd && start via systemd
2. check upstart && start via upstart
3. check update-rc.d
    1. register via update-rc.d
    2. (check service && start via service) || start via /etc/init.d

same goes for remove

**Comment by heartsucker** [comment]
I thought about adding this too, but the two distros I was targeting when writing this, Debian 8 and Ubuntu 14.04, both use systemd or Upstart as their init service. In face, Debian 9 and Ubuntu 16.04 are both systemd. System V init support feels dated, and I'm not sure this is necessary. Are there any Debian based systems that don't use systemd or Upstart?

**Comment by kmohrf** [comment]
they both use it as default but system v - like init is still supported. on embedded systems init scripts are quite common and i know at least a handful of admins who still prefer init scripts over systemd (actually most of them prefer unit over init files but consider systemd a security and/or ecosystem threat).
i added this one because i actually needed it in case that helps with your decision :)

File: templates/postinst [hunk 1]
```diff
@@ -56,13 +56,20 @@ start_service () {
...
     fi
   elif hash start 2> /dev/null; then
     if [[ "$init_type" == 'auto' || "$init_type" == 'upstart' ]]; then
       start "$service_name" || echo "$service_name could not be registered or started" # using `|| echo` to ensure script exits with 0. Failure to start is not failure to install
     fi
+  elif hash update-rc.d 2> /dev/null; then
+    if [[ "$init_type" == 'auto' || "$init_type" == 'sysv' ]]; then
+      {
+        update-rc.d "$service_name" defaults && \
+        "/etc/init.d/$service_name"
```
Comment (heartsucker) [inline templates/postinst]: I believe you are missing a `start` as an argument to the script

Comment (kmohrf) [inline templates/postinst]: fixed

**Comment by kmohrf** [comment]
any comments regarding [service registration & start/stop](https://github.com/heartsucker/node-deb/pull/48#issuecomment-284743972)

**Comment by heartsucker** [comment]
@kmohrf Upstart registers the service via the run levels, so only System V and systemd need to explicitly enable the service. In face, if you've ever done a `systemctl disable foo` where `foo` only has an Upstart config, you'll get a `foo.conf.override` that has new run levels defined.

Anyway. I'm merging this and will get this published. In the mean time, I still have to do more review on your other PR.

**Pull request merged**